src/inc/volatile.h

```diff
@@ -120,7 +120,7 @@ T VolatileLoad(T const * pt)
     STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
 
 #ifndef DACCESS_COMPILE
-#if defined(_ARM64_) && defined(__GNUC__)
+#if defined(_ARM64_) && defined(__GNUC__) && !(defined(__BIONIC__))
```

Comment:
@janvorli Thanks for the tip, but unfortunately the same error is thrown with the cast.  Not sure if I mentioned this earlier but I only have a rudimentary knowledge of C/C#.

What version of clang is used to build Linux? The NDK uses version 3.8